### PR TITLE
Recipe Krb5LoginModuleClass

### PR DESCRIPTION
**Pull request by BhavanaPidapa**
## What's changed?
image

## What's your motivation?
I used org.openrewrite.java.ChangeType for the recipe Krb5LoginModuleClass

## Anyone you would like to review specifically?
@cjobinabo , @timtebeek 

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Hi @BhavanaPidapa & welcome to the project! I'm wondering if these recipes might be a better fit in [openrewrite/rewrite-liberty](https://github.com/openrewrite/rewrite-liberty/), given that it seems very specific to IBM. I'm not sure if just any user of `com.sun.security` would want to migrate to `com.ibm.security`. It might even make sense to develop a `rewrite-ibm` module that uses `rewrite-migrate-java` and adds the specific IBM steps there.  Any thoughts around that?

**Comment by BhavanaPidapa** [comment]
> Hi @BhavanaPidapa & welcome to the project! I'm wondering if these recipes might be a better fit in [openrewrite/rewrite-liberty](https://github.com/openrewrite/rewrite-liberty/), given that it seems very specific to IBM. I'm not sure if just any user of `com.sun.security` would want to migrate to `com.ibm.security`. It might even make sense to develop a `rewrite-ibm` module that uses `rewrite-migrate-java` and adds the specific IBM steps there. Any thoughts around that?

Sure Tim, I will check with Chuka and get back on this.

**Comment by cjobinabo** [comment]
Hey @timtebeek, are you okay with moving this recipe to the https://github.com/openrewrite/rewrite-migrate-java/blob/main/src/main/resources/META-INF/rewrite/ibm-java.yml file?

**Comment by cjobinabo** [comment]
Actually, perhaps a new list for moving from older IBM runtimes to none IBM implementations may make more sense. We wouldn't want to move this recipe to the liberty repo because that repo was meant to handle migration issues encountered when moving from our tWAS app server to our Liberty app server.

**Comment by timtebeek** [comment]
Indeed then sounds like rewrite-liberty isn't a good fit; but a separate recipe inside ibm-java.yml might make most sense then.

**Comment by cjobinabo** [comment]
That sounds good to me.

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2022 the original author or authors.
+# Copyright 2024 the original author or authors.
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:2]:
```suggestion
# Copyright 2022 the original author or authors.
```

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 2]
```diff
@@ -57,16 +57,17 @@ recipeList:
   - org.openrewrite.java.migrate.cobertura.RemoveCoberturaMavenPlugin
   - org.openrewrite.java.migrate.wro4j.UpgradeWro4jMavenPluginVersion
   - org.openrewrite.java.migrate.UpgradeBuildToJava11
-# Disabled due to null safety issues in the current implementation
-# https://github.com/openrewrite/rewrite-migrate-java/issues/250
-#  - org.openrewrite.java.migrate.util.JavaUtilAPIs
+  # Disabled due to null safety issues in the current implementation
+  # https://github.com/openrewrite/rewrite-migrate-java/issues/250
+  #  - org.openrewrite.java.migrate.util.JavaUtilAPIs
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:62]:
```suggestion
# Disabled due to null safety issues in the current implementation
# https://github.com/openrewrite/rewrite-migrate-java/issues/250
#  - org.openrewrite.java.migrate.util.JavaUtilAPIs
```

File: src/main/resources/META-INF/rewrite/java-version-11.yml [hunk 3]
```diff
@@ -57,16 +57,17 @@ recipeList:
...
   - org.openrewrite.java.migrate.util.OptionalNotPresentToIsEmpty
   - org.openrewrite.java.migrate.util.OptionalNotEmptyToIsPresent
   - org.openrewrite.java.migrate.util.OptionalStreamRecipe
   - org.openrewrite.java.migrate.InternalBindContextFactory
   - org.openrewrite.java.migrate.RemovedSecurityManagerMethods
   - org.openrewrite.java.migrate.UpgradePluginsForJava11
 
+
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/java-version-11.yml:70]:
```suggestion
```

File: src/test/java/org/openrewrite/java/migrate/Java8ToJava11Test.java [hunk 4]
```diff
@@ -32,9 +33,22 @@ public void defaults(RecipeSpec spec) {
           .parser(JavaParser.fromJavaVersion()
             .classpathFromResources(new InMemoryExecutionContext(), "sun.internal.new"))
           .recipe(Environment.builder().scanRuntimeClasspath("org.openrewrite.java.migrate").build()
-            .activateRecipes("org.openrewrite.java.migrate.Java8toJava11"));
+            .activateRecipes("org.openrewrite.java.migrate.Java8toJava11","org.openrewrite.java.migrate.IBMJDKtoOracleJDK"));
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/Java8ToJava11Test.java:36]:
```suggestion
            .activateRecipes("org.openrewrite.java.migrate.Java8toJava11", "org.openrewrite.java.migrate.IBMJDKtoOracleJDK"));
```

File: src/test/java/org/openrewrite/java/migrate/Java8ToJava11Test.java [hunk 5]
```diff
@@ -83,27 +97,65 @@ void bar() {
...
                     factory.hashCode();
                 }
+
+              }
+              """
+          )
+        );
+    }
+
+    @DocumentExample
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/Java8ToJava11Test.java:125]:
```suggestion
```

**Review by timtebeek — COMMENTED** [review]
Quick few suggestions to revert accidental changes and improve formatting

**Review by cjobinabo — APPROVED** [review]
looks good to me

**Review by timtebeek — APPROVED** [review]
Reworked the tests a bit to introduce a separate IBMJDKtoOracleJDKTest, as Java8To11Test didn't have much to do with this separate recipe. Otherwise OK to merge, thanks!

**Pull request merged**